parcels/particleset/particlesetnodes.py

```diff
@@ -0,0 +1,1389 @@
+
+            if verbose_progress is None and time_module.time() - walltime_start > 10:
+                # Showing progressbar if runtime > 10 seconds
+                if output_file:
+                    logger.info('Temporary output files are stored in %s.' % output_file.tempwritedir_base)
+                    logger.info('You can use "parcels_convert_npydir_to_netcdf %s" to convert these '
+                                'to a NetCDF file during the run.' % output_file.tempwritedir_base)
+                pbar = self._create_progressbar_(_starttime, endtime)
+                verbose_progress = True
+
+            if dt > 0:
+                time = min(next_prelease, next_input, next_output, next_movie, next_callback, endtime)
+            else:
+                time = max(next_prelease, next_input, next_output, next_movie, next_callback, endtime)
+            self._kernel.execute(self, endtime=time, dt=dt, recovery=recovery, output_file=output_file, execute_once=execute_once)
+            if abs(time-next_prelease) < tol:
+                ngrids = self.fieldset.gridset.size if self.fieldset is not None else 0
+                add_iter = 0
+                while add_iter < len(self.repeatlon):
+                    gen_id = None if self.repeatpid is None or type(self.repeatpid) not in [list, tuple, dict, np.ndarray] else self.repeatid[add_iter]
+                    lon = self.repeatlon[add_iter]
+                    lat = self.repeatlat[add_iter]
+                    pdepth = self.repeatdepth[add_iter]
+                    ptime = time if self.release_starttime is None else self.release_starttime
+                    pid = self._idgen.nextID(lon, lat, pdepth, ptime) if gen_id is None else gen_id
+                    pdata = self.repeatpclass(lon, lat, pid=pid, ngrids=ngrids, depth=pdepth, time=ptime)
+                    pdata.dt = dt
+                    # Set other Variables if provided
+                    for kwvar in self.repeatkwargs:
+                        if isinstance(kwvar, Field):
+                            continue
+                        if not hasattr(pdata, kwvar):
+                            raise RuntimeError('Particle class does not have Variable %s' % kwvar)
+                        setattr(pdata, kwvar, self.repeatkwargs[kwvar][add_iter])
+                    self.add(pdata)
+                    add_iter += 1
+                next_prelease += self.repeatdt * np.sign(dt)
+            if abs(time-next_output) < tol or execute_once:
+                for fld in self._fieldset.get_fields():
+                    if hasattr(fld, 'to_write') and fld.to_write:
+                        if fld.grid.tdim > 1:
+                            raise RuntimeError('Field writing during execution only works for Fields with one snapshot in time')
+                        fldfilename = str(output_file.name).replace('.nc', '_%.4d' % fld.to_write)  # what does this do ? the variable is boolean, then it's increased - what-the-frog ...
```

Comment:
This is the hack we discussed before. Perhaps open an Issue to fix this and remove the comment here?

Reply:
ups - sorry, copy-mistake with the comment.